src/protocols/aeternity/composables/aeNames.ts

```diff
@@ -0,0 +1,290 @@
+import {
+  ref,
+  watch,
+} from 'vue';
+import {
+  fetchAllPages,
+  fetchJson,
+  handleUnknownError,
+} from '@/utils';
+import { useMiddleware } from '@/composables/middleware';
+import { useAccounts } from '@/composables/accounts';
+import { Encoded } from '@aeternity/aepp-sdk';
+import { useTransactionList } from '@/composables/transactionList';
+import type {
+  ChainName,
+  IDefaultComposableOptions,
+  IName,
+  ITransaction,
+} from '@/types';
+import {
+  AUTO_EXTEND_NAME_BLOCKS_INTERVAL,
+  PROTOCOL_AETERNITY,
+  UPDATE_POINTER_ACTION,
+} from '@/constants';
+import { useAeSdk } from '@/composables/aeSdk';
+import { isInsufficientBalanceError } from '@/protocols/aeternity/helpers';
+import {
+  IAddressNamePair,
+  IAuction,
+  IAuctionBid,
+} from '@/types';
+import { useAeNetworkSettings } from '@/protocols/aeternity/composables/aeNetworkSettings';
+import { useModals } from '@/composables/modals';
+import { tg } from '@/store/plugins/languages';
+import {
+  createPollingBasedOnMountedComponents,
+  useStorageRef,
+} from '@/composables/composablesHelpers';
+
+const POLLING_INTERVAL = 10000;
+
+interface IUpdatePointerParams {
+  name: ChainName;
+  address?: any;
+  type?: keyof typeof UPDATE_POINTER_ACTION;
+}
+
+interface IAuctionEntryParams {
+  name: ChainName;
+  expiration: number;
+  bids: IAuctionBid[];
+}
+
+const ownedNames = useStorageRef<IName[]>([], 'owned-names');
+const pendingAutoExtendNames = ref<ChainName[]>([]);
+const areNamesFetching = ref(false);
+const defaults = ref<Record<string, ChainName>>({});
+const preferred = ref<Record<string, string>>({});
```

Comment:
To make it more readable can you please rename this constant to something like `preferredNamesRegistry`?
Also it would be nice if you could add a comment about what the `string` stands for. And a general comment what `preferred` means.

Reply:
After some digging and explanation I see that the name `preferred` is misleading as this state holds the names of the accounts that are not local. So I think more suitable name would be for example `external` (`externalNamesRegistry`).